causalpy/pymc_experiments.py

```diff
@@ -54,12 +54,23 @@ class TimeSeriesExperiment(ExperimentalDesign):
     def __init__(
         self,
         data: pd.DataFrame,
-        treatment_time: int,
+        treatment_time: int | float | pd.Timestamp,
         formula: str,
         model=None,
         **kwargs,
     ) -> None:
         super().__init__(model=model, **kwargs)
+
+        # Input validation
```

Comment:
I suggest we add this into a private function

```python
def _validate_input_indices(self, ...):
    ...
```
and simply call it inside the `__init __` method.

Reply:
God idea. 

We already have some input validation in the other classes, so I'm thinking a more general input validation method.

Reply:
Done